src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -212,5 +214,35 @@ public override void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions
                     break;
             }
         }
+
+        private JsonEncodedText FormatEnumValue(string value)
+        {
+            string converted;
+
+            if (!value.Contains(ValueSeparator))
+            {
+                converted = _namingPolicy.ConvertName(value);
+            }
+            else
+            {
+                // todo: optimize implementation here by leveraging https://github.com/dotnet/runtime/issues/934.
+                string[] enumValues = value.Split(
+#if BUILDING_INBOX_LIBRARY
+                    ValueSeparator
+#else
+                    new string[] { ValueSeparator }, StringSplitOptions.None
+#endif
+                    );
+
+                for (int i = 0; i < enumValues.Length; i++)
+                {
+                    enumValues[i] = _namingPolicy.ConvertName(enumValues[i]);
+                }
+
+                converted = string.Join(ValueSeparator, enumValues);
+            }
+
+            return JsonEncodedText.Encode(converted);
```

Comment:
Use the encoder specified on options?

` return JsonEncodedText.Encode(converted, options.Encoder);`